pkg/config/config.go

```diff
@@ -10,28 +10,14 @@ import (
 type DaemonConfig struct {
 	PeriodicUpdate int `env:"PERIODIC_UPDATE" envDefault:"5"` // Interval between every check for the added and deleted pods
 	GuidPool       GuidPoolConfig
-	SubnetManager  SubnetManagerPluginConfig
+	Plugin         string `env:"PLUGIN"` // Subnet manager plugin name
```

Comment:
nit: to make things more explicit perhaps rename the env variable to SM_PLUGIN ?